src/flow.c

```diff
@@ -452,6 +452,10 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
             f->flags &= ~FLOW_PROTO_DETECT_TS_DONE;
             p->flags |= PKT_PROTO_DETECT_TS_DONE;
         }
+        if (f->macset != NULL && p->ethh != NULL) {
+            MacSetAdd(f->macset, p->ethh->eth_dst, TOSERVER);
```

Comment:
If we always have 2 mac addresses, shouldn't we handle things in a single call?

Reply:
Definitely. Will look into it.